install_config/configure_kerberos_for_sso.adoc

```diff
@@ -0,0 +1,174 @@
+two component principal where the first component is the string `HTTP` (all
+capitals) and the second part is the fully qualified host name (based on the
+chosen DNS name for the proxy). A SPN for this service is normally represented
+in strong form like `HTTP/hostname@REALM` where the realm is generally the
+uppercase name of the domain controlled by your KDC.
+
+Assuming a domain named krb.example.com where the KDC resides we choose a DNS
+name like auth-proxy.krb.example.com. The resulting SPN we need to create in
+the kerberos realm is: `HTTP/auth-proxy.krb.example.com@KRB.EXAMPLE.COM`
+
+Once the SPN is created assign a random key to it and extract a keytab, export
+it to a file and securely transfer it to the proxy server.
+[IMPORTANT]
+====
+The keytab contains the secret key of the service and must be transfered
+securely to the target system without leving copies in files accessible by
+other users. Access to those keys allows a complete authentication bypass
+and jeopardizes the security of your installation.
+====
+
+We assume the cluster uses the public name: `cluster.krb.example.com`
+
+We also need a TLS certificate for serving content for the chosen DNS
+name. The certificate must have a SAN for the name: `auth-proxy.krb.example.com`
+or a wildcard name for the whole domain.
+
+Finally we also need a 'client' certificate that the proxy uses to proxy to the
+openshift cluster. A client certificate is a certificate with extended key
+usage property set to `TLS Web client authentication`.
+[IMPORTANT]
+====
+This certificate is critical as it will be fully trusted by the cluster to
+provide user identity data. A compromise of this certificate will allow an
+attacker to impersonate any user in the cluster.
+====
+In our example we create a certificate with the subject: `CN=oauth-proxy-client`
+
+===== Configuring Apache
+
+Two modules are required for a successful configuration: mod_request and
+mod_auth_gssapi, make sure the following configuration lines are uncommented
+in the proper configurtion file (usually something like :
+
+----
+LoadModule request_module modules/mod_request.so
+LoadModule auth_gssapi_module modules/mod_auth_gssapi.so
+----
+
+The next step is to add virtual host directives to allow Apache to respond to
+queries made to the DNS name we chose earlier:
+
+----
+# Nothing needs to be served over HTTP.  This virtual host simply redirects to
+# HTTPS.
+<VirtualHost *:80>
+  DocumentRoot /var/www/html
+  RewriteEngine              On
+  RewriteRule     ^(.*)$     https://%{HTTP_HOST}$1 [R,L]
+</VirtualHost>
+
+<VirtualHost *:443>
+  # ServerName needs to match the DNS name we chose and the SAN of the serving
+  # certificate that was generated as well as the hostname of the Service
+  # Principal Name created to obtain the keys in the keytab.
+  ServerName auth-proxy.krb.example.com
+
+  DocumentRoot /var/www/html
+  SSLEngine On
+  SSLCertificateFile /etc/pki/tls/certs/auth-proxy.krb.example.com.crt
```

Comment:
Which certs are these?

Reply:
The ones we mention you need as prerequisite